src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Main {
+
+    public static void main(String[] args) {
+        final int arrayLength = 6;
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < arrayLength; i++) {
+            if (i < arrayLength / 2) {
+                figures[i] = new FigureSupplier().getRandomFigure();
+                figures[i].draw();
+            } else {
+                figures[i] = new FigureSupplier().getDefaultFigure();
```

Comment:
Consider creating a single instance of `FigureSupplier` outside the loop to avoid unnecessary instantiation within each iteration. This will improve performance and readability.